### DISCUSSION TITLE
CAN bus Log file in CSV format - How to convert to a MDF4 file

### DISCUSSION THREAD

**Discussion by regra-uri** [danielhrisca/asammdf#1037] [Q&A]
Hey guys! I have a student trying to convert a CSV log file to a MDF4 file, following the correct file structure. If someone could help providing some example or python script to do so, or any help! Thank you all!
Here is how the CSV file is formated based on OBD response pids:
TimestampEpoch;BusChannel;ID;IDE;DLC;DataLength;Dir;EDL;BRS;DataBytes
1718284701.177;1;7E8;0;8;8;0;0;0;03410400AAAAAAAA

**Comment by regra-uri** [comment]
OK, I implemented something in python but validating in the Vector MDF validator, I got many warnings about gap between current block and next block by 8 bytes. I understood that the Channel Hierarchy is not implemented yet, then it is not possible. I will try to do it or use the turboLab lib. Tks guys!

**Comment by danielhrisca** [comment]
You can get some insights from this code

https://github.com/hardbyte/python-can/blob/main/can/io/mf4.py
